pypeit/images/combineimage.py

```diff
@@ -144,32 +196,43 @@ def run(self, bias=None, flatimages=None, ignore_saturation=False, sigma_clip=Tr
             print(msgs.indent() + '-'*maxlen + "  " + '-'*maxlmp)
 
         # Coadd them
-        weights = np.ones(nimages)/float(nimages)
-        img_list = [img_stack]
         var_stack = utils.inverse(ivar_stack)
-        var_list = [var_stack, rn2img_stack]
         if combine_method == 'weightmean':
-            img_list_out, var_list_out, gpm, nused = combine.weighted_combine(
-                weights, img_list, var_list, (mask_stack == 0),
-                sigma_clip=sigma_clip, sigma_clip_stack=img_stack, sigrej=sigrej, maxiters=maxiters)
+            weights = np.ones(self.nfiles, dtype=float)/self.nfiles
+            img_list = [img_stack]
+            var_list = [var_stack, rn2img_stack]
+            img_list_out, var_list_out, gpm, nused \
+                    = combine.weighted_combine(weights, img_list, var_list, mask_stack==0,
+                                               sigma_clip=sigma_clip, sigma_clip_stack=img_stack,
+                                               sigrej=sigrej, maxiters=maxiters)
         elif combine_method == 'median':
-            # TODO This median combine should be using the mask!!!
-            img_list_out = [np.median(img_stack, axis=0)]
-            var_list_out = [np.median(var_stack, axis=0)]
-            var_list_out += [np.median(rn2img_stack, axis=0)]
-            gpm = np.ones_like(img_list_out[0], dtype='bool')
+            bpm_stack = mask_stack > 0
+            gpm = np.sum(np.logical_not(bpm_stack), axis=0)
+            img_list_out = [np.ma.median(np.ma.MaskedArray(img_stack, mask=bpm_stack),
+                                         axis=0).filled(0.)]
+            var_list_out = [np.ma.median(np.ma.MaskedArray(var_stack, mask=bpm_stack),
+                                         axis=0).filled(0.)]
+            var_list_out += [np.ma.median(np.ma.MaskedArray(rn2img_stack, mask=bpm_stack),
+                                          axis=0).filled(0.)]
         else:
+            # NOTE: Given the check at the beginning of the function, the code
+            # should *never* make it here.
             msgs.error("Bad choice for combine.  Allowed options are 'median', 'weightmean'.")
 
         # Build the last one
         final_pypeitImage = pypeitimage.PypeItImage(img_list_out[0],
                                                     ivar=utils.inverse(var_list_out[0]),
+                            # TODO: Why is this `pypeitImage.bpm` and not
+                            # `np.logical_not(gpm)`?
                                                     bpm=pypeitImage.bpm,
                                                     rn2img=var_list_out[1],
+                            # TODO: The mask can be set for a bunch of reasons.
+                            # Why identify all of the masked pixels as cosmic rays?
                                                     crmask=np.logical_not(gpm),
```

Comment:
Yeah so the issue was that I could not figure out how to combine the bits in weighted_combine. Combining bits in averaged images is a bit of a subtle point, but I think for CRs it could conceivably be done.